pages/api/bitbucket/callbacks/webhook.ts

```diff
@@ -61,20 +76,28 @@ const webhookHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 
 		const result: string | null = await publishMessage(installId, data, msgType)
 		.catch((error) => {
+			const eventProperties = { ...event_properties, topic_name: installId };
+			rudderStackEvents.track("absent", "", 'bitbucket-webhook', { type: 'publish-webhook-message', eventStatusFlag: 0, eventProperties });	
 			console.error('[webookHandler] Failed to publish message:', error);
 			failedCount++;
 			return null;
 		});
 
 		if (result === null) continue;
 
+		const eventProperties = { ...event_properties, topic_name: installId };
+		rudderStackEvents.track("absent", "", 'bitbucket-webhook', { type: 'publish-webhook-message', eventStatusFlag: 1, eventProperties });
 		console.info("[webookHandler] Sent message to pubsub for ", installId, result);
 	}
 
 	// Determine the response status code based on the number of failures
 	if (failedCount > 0) {
+		const eventProperties = { ...event_properties, response_status: 500, failed_count: failedCount};
+		rudderStackEvents.track("absent", "", 'bitbucket-webhook', { type: 'HTTP-500', eventStatusFlag: 0, eventProperties });
 		res.status(500).json({ error: `Failed to publish ${failedCount} messages to Pub/Sub` });
 	} else {
+		const eventProperties = { ...event_properties, response_status: 200, failed_count: failedCount };
```

Comment:
Please add the `topicName` in the event properties. It will help me join values across different tables to understand whose repositories were getting processed.

Reply:
plus any way to know whose repos are being processed, events have been handled in the for loop itself where I add `topic_name` for a success case as well as a failure case too.